code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -699,76 +699,100 @@
 ////////////////
 /* Ammo Boxes */
 ////////////////
-/obj/item/storage/box/slug
+
+// Empty type
+// All these can only hold their special ammo types and automatically are filled with them
+/obj/item/storage/fancy/shell
+	icon = 'icons/obj/shell_boxes.dmi'
+	storage_slots = 8
+	appearance_flags = parent_type::appearance_flags | KEEP_TOGETHER
+	can_hold = list(/obj/item/ammo_casing/shotgun)
+
+/obj/item/storage/fancy/shell/update_icon_state()
+	icon_state = "open"
+
+/obj/item/storage/fancy/shell/update_overlays()
+	. = ..()
+	var/list/cached_contents = contents
+	for(var/index in 1 to length(cached_contents))
+		var/obj/shell = cached_contents[index]
+		var/image/I = image(icon, src, initial(shell.icon_state))
+		I.pixel_x = 3 * (round((index - 1) / 2))
+		I.pixel_y = -4 * ((index + 1) % 2)
+		. += I
+
+	. += "shell_box_front" // need to add another overlay to prevent from other overlays from showing on top
+
+/obj/item/storage/fancy/shell/tranquilizer
+	name = "ammunition box (Tranquilizer darts)"
+	desc = "A small box capable of holding seven shotgun shells."
+	icon_state = "tranqbox"
+
+/obj/item/storage/fancy/shell/tranquilizer/populate_contents()
+	for(var/i in 1 to 8)
+		new /obj/item/ammo_casing/shotgun/tranquilizer(src)
```

Comment:
Make this a var on `/obj/item/storage/fancy/shell`, something like `shell_type`, null by default.
```
/obj/item/storage/fancy/shell/populate_contents()
	if(!shell_type)
		return
	for(var/i in 1 to storage_slots)
		new shell_type(src)
```